File: swapi-angular/src/app/api/models/FilmDTO.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { SearchItemDTO } from './SearchItemDTO';

export type FilmDTO = {
    episode_id?: string;
    opening_crawl?: string;
    release_date?: string;
    created?: string;
    director?: string;
    edited?: string;
    producer?: string;
    title?: string;
    url?: string;
    characters?: Array<string>;
    characterItems?: Array<SearchItemDTO>;
    planets?: Array<string>;
    planetItems?: Array<SearchItemDTO>;
    species?: Array<string>;
    speciesItems?: Array<SearchItemDTO>;
    starships?: Array<string>;
    starshipItems?: Array<SearchItemDTO>;
    vehicles?: Array<string>;
    vehicleItems?: Array<SearchItemDTO>;
};

File: swapi-server/src/main/java/de/lkaktus/entities/FilmDTO.java
package de.lkaktus.entities;

import com.fasterxml.jackson.annotation.JsonProperty;

public record FilmDTO(
        String created,
        String director,
        String edited,
        @JsonProperty("episode_id") String episodeID,
        @JsonProperty("opening_crawl") String openingCrawl,
        String producer,
        @JsonProperty("release_date") String releaseDate,
        String title,
        String url,
        String [] characters,
        SearchItemDTO [] characterItems,
        String [] planets,
        SearchItemDTO [] planetItems,
        String [] species,
        SearchItemDTO [] speciesItems,
        String [] starships,
        SearchItemDTO [] starshipItems,
        String [] vehicles,
        SearchItemDTO [] vehicleItems
) {

}
